new/analyse/prepare-changelogs.ts

```diff
@@ -0,0 +1,85 @@
+// import { Message } from './../types';
+import type {
+  Module,
+  PackageAfterDetermineVersion,
+  PackageAfterPrepareChangelogs,
+  PackageChangelog,
+  MainChangelog,
+  ChangelogMessage,
+} from '../types';
+import { clone } from 'ramda';
+import fs from 'fs';
+
+import { logger } from '../helpers/logger';
+import { join } from 'path';
+
+import { getReleasablePackages } from './adapt-dependencies/get-releasable-packages';
+
+const { error, log } = logger('[analyse] prepare changelogs');
+
+export const prepareChangelogs: Module = (env) => {
+  if (!env.packages) {
+    const errorMessage = '"packages" attribute not found on env config object';
+    error(errorMessage);
+    throw new Error(errorMessage);
+  }
+
+  const clonePackages = clone(env.packages);
+  const releasablePackages = getReleasablePackages(clonePackages);
+
+  const changelogDate = new Date().toISOString();
+  let mainChangeLogContent: MainChangelog = { [changelogDate]: [] };
+
+  releasablePackages.forEach((packageName) => {
+    const currentPackage = clonePackages[
+      packageName
+    ] as PackageAfterDetermineVersion;
+    log(`preparing changelog messages for ${packageName} `);
+
+    const changelogFile = join(currentPackage.path, 'changelog.json');
+    const version = currentPackage.incrementedVersion;
+    const messages: ChangelogMessage[] = currentPackage.messages.map(
```

Comment:
Oh. Found one more thing - sorry, didn't catch it earlier. There are also `relatedMessages`. They have to be merged into that array as well.

Reply:
done